src/data/list/chain.lean

```diff
@@ -72,6 +72,29 @@ theorem chain_map_of_chain {S : β → β → Prop} (f : α → β)
   (p : chain R a l) : chain S (f a) (map f l) :=
 (chain_map f).2 $ p.imp H
 
+theorem chain_pmap_of_chain {S : β → β → Prop} {p : α → Prop}
+  {f : Π a, p a → β}
+  (H : ∀ a b ha hb, R a b → S (f a ha) (f b hb))
+  {a : α} {l : list α}
+  (hl₁ : chain R a l) (ha : p a) (hl₂ : ∀ a ∈ l, p a) :
+  chain S (f a ha) (list.pmap f l hl₂) :=
+begin
+  induction l with lh lt l_ih generalizing a,
+  { simp },
+  { simp [H _ _ _ _ (rel_of_chain_cons hl₁), l_ih _ (chain_of_chain_cons hl₁)] }
+end
+
+theorem chain_of_chain_pmap {S : β → β → Prop} {p : α → Prop}
+  (f : Π a, p a → β) (l : list α) (hl₁ : ∀ a ∈ l, p a)
```

Comment:
Here too. At the very least `l` should be implicit, it appears in later hypotheses and in the conclusion.

Reply:
Should the same apply to `chain_of_chain_map` above as well?

Reply:
Yes, it looks like `f` can be made implicit in several of the other lemmas near there as well. I don't think it necessarily has to be done in this PR though.